src/rcms/fm/app/level1/HCALlevelTwoFunctionManager.java

```diff
@@ -67,12 +75,44 @@ public String getSupervisorErrorMessage() {
             // none is a place holder on xDAQ side for RCMS to safely look at the infospace at anytime.
             if(!errAppName.equals("none")){
               MapT<StringT> errMap = new MapT<StringT>();
+              StringT errAppURI    = new StringT("");
+              StringT errAppCrate  = new StringT("");
               // Associate name of app to err_message of app by vector position
-              //Name_err_map.put( "("+partition+") "+ errAppName, errAppMsgVector.get(  errAppNameVector.indexOf( errAppName ))) ;
               errMap.put( "timestamp", new StringT(getTimestampString()));
-              errMap.put( "app", new StringT("("+partition+") "+ errAppName));
               errMap.put("message", errAppMsgVector.get(  errAppNameVector.indexOf( errAppName ))) ;
-              
+
+              // Look for the URI of the app name in allHandled app name vector 
+              for (StringT HandledAppName : AllHandledAppNameVector){
+                if (HandledAppName.equals(errAppName)){
+                  errAppURI = AllHandledAppURIVector.get(  AllHandledAppNameVector.indexOf( HandledAppName ));
+                  errMap.put("URI", errAppURI) ;
+                }
+              }
+              // Extract crate number from URI
+              if (errAppURI.length()>0){
+                // e.g. URI = http://hcal904daq01.cms904:39100/urn:xdaq-application:lid=53
+                String errAppPortNumber  = errAppURI.getString().split("/")[2].split(":")[1];
+                // 3 digit in port number = isCrate boolean
+                if (String.valueOf(errAppPortNumber.charAt(2)).equals("1")){
+                  errAppCrate = new StringT(errAppPortNumber.substring(3));
```

Comment:
I don't like this method for identifying the crate number. It will only work if the confmagik accords to the port numbering convention, which is not the case at all teststands.

I suggest instead querying the hcalCrate application for it's crate number.

Reply:
1) The crate executive could crash, so the infospace query will not work in that case
2) I don't like getting sibling application if not 100% necessary because that's the only thing preventing us from using light configuration. 
3) I don't see why teststands cannot follow the p5 conventions on port numbers.

Reply:
IIRC, @sethcooper mentioned that he asked Hannes about the light config problem with finding sibling apps in light configs, and that Hannes replied to the effect that light configs _should_ support that. So probably that should be fixed on the framework side.

Anyhow I think using light configs is usually a bad idea so I would support using the the sibling app method.

Reply:
With regards to this point:
> I don't see why teststands cannot follow the p5 conventions on port numbers.
From my experience at FNAL and helping the TIFR group, the absolute hardest part of maintaining a teststand is changing things at the confmagik level. Now FNAL and TIFR are both using (or will soon use) configs that do not accord with the P5 convention, and I don't think they will be updated by anyone in the foreseeable future.